homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,284 @@
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER_NAME
+from .coordinator import CompitConfigEntry, CompitDataUpdateCoordinator
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+ATTR_TEMPERATURE = "temperature"
+PARAM_PRESET_MODE = "__trybpracytermostatu"
+PARAM_FAN_MODE = "__trybaero"
+PARAM_HVAC_MODE = "__trybpracyinstalacji"
+PARAM_CURRENT_TEMPERATURE = "__tpokojowa"
+PARAM_TARGET_TEMPERATURE = "__tpokzadana"
+PARAM_SET_TARGET_TEMPERATURE = "__tempzadpracareczna"
+CLIMATE_DEVICE_CLASS = 10
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+
+class CompitHVACMode(Enum):
+    """Enum for available HVAC modes."""
+
+    HEAT = 0
+    OFF = 1
+    COOL = 2
+
+
+COMPIT_MODE_MAP = {
+    CompitHVACMode.COOL: HVACMode.COOL,
+    CompitHVACMode.HEAT: HVACMode.HEAT,
+    CompitHVACMode.OFF: HVACMode.OFF,
+}
+
+HVAC_MODE_TO_COMPIT_MODE = {v: k for k, v in COMPIT_MODE_MAP.items()}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: CompitConfigEntry,
+    async_add_devices: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the CompitClimate platform from a config entry."""
+
+    coordinator: CompitDataUpdateCoordinator = entry.runtime_data
+    async_add_devices(
+        [
+            CompitClimate(
+                coordinator,
+                device,
+                device_definition.parameters,
+                device_definition.name,
+            )
+            for gates in coordinator.gates
+            for device in gates.devices
+            if (
+                device_definition := next(
+                    (
+                        definition
+                        for definition in coordinator.device_definitions.devices
+                        if definition.code == device.type
+                    ),
+                    None,
+                )
+            )
+            is not None
+            if (device_definition.device_class == CLIMATE_DEVICE_CLASS)
+        ]
+    )
+
+
+class CompitClimate(CoordinatorEntity[CompitDataUpdateCoordinator], ClimateEntity):
+    """Representation of a Compit climate device."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.COOL, HVACMode.OFF]
+    _attr_name = None
+    _attr_has_entity_name = True
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.FAN_MODE
+        | ClimateEntityFeature.PRESET_MODE
+    )
+
+    def __init__(
+        self,
+        coordinator: CompitDataUpdateCoordinator,
+        device: Device,
+        parameters: list[Parameter],
+        device_name: str,
+    ) -> None:
+        """Initialize the climate device."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{device.label}_{device.id}"
```

Comment:
The label seems to be the name, but we should not include the name in the unique id as that can change